subsys/bluetooth/controller/hci_cmd_put.c

```diff
@@ -0,0 +1,851 @@
+		*param_length_out += sizeof(sdc_hci_cmd_vs_zephyr_write_tx_power_return_t);
+		return sdc_hci_cmd_vs_zephyr_write_tx_power((void *)cmd_params,
+							    (void *)event_out_params);
+	case SDC_HCI_OPCODE_CMD_VS_LLPM_MODE_SET:
+		return sdc_hci_cmd_vs_llpm_mode_set((void *)cmd_params);
+	case SDC_HCI_OPCODE_CMD_VS_CONN_UPDATE:
+		return sdc_hci_cmd_vs_conn_update((void *)cmd_params);
+	case SDC_HCI_OPCODE_CMD_VS_CONN_EVENT_EXTEND:
+		return sdc_hci_cmd_vs_conn_event_extend((void *)cmd_params);
+	case SDC_HCI_OPCODE_CMD_VS_QOS_CONN_EVENT_REPORT_ENABLE:
+		return sdc_hci_cmd_vs_qos_conn_event_report_enable((void *)cmd_params);
+	case SDC_HCI_OPCODE_CMD_VS_EVENT_LENGTH_SET:
+		return sdc_hci_cmd_vs_event_length_set((void *)cmd_params);
+	case SDC_HCI_OPCODE_CMD_VS_ZEPHYR_WRITE_BD_ADDR:
+		return sdc_hci_cmd_vs_zephyr_write_bd_addr((void *)cmd_params);
+	default:
+		return BT_HCI_ERR_UNKNOWN_CMD;
+	}
+}
+
+
+static void cmd_put(uint8_t *cmd_in, uint8_t * const raw_event_out)
+{
+	uint8_t status;
+	uint16_t opcode = *(uint16_t *)cmd_in;
+	uint8_t return_param_length = CMD_COMPLETE_MIN_SIZE;
+
+	switch (BT_OGF(opcode)) {
+#if defined(CONFIG_BT_CONN)
+	case BT_OGF_LINK_CTRL:
+		status = link_control_cmd_put(cmd_in);
+		break;
+#endif
+	case BT_OGF_BASEBAND:
+		status = controller_and_baseband_cmd_put(cmd_in,
+							 raw_event_out,
+							 &return_param_length);
+		break;
+	case BT_OGF_INFO:
+		status = info_param_cmd_put(cmd_in,
+					    raw_event_out,
+					    &return_param_length);
+		break;
+	case BT_OGF_STATUS:
+		status = status_param_cmd_put(cmd_in,
+					      raw_event_out,
+					      &return_param_length);
+		break;
+	case BT_OGF_LE:
+		status = le_controller_cmd_put(cmd_in,
+					       raw_event_out,
+					       &return_param_length);
+		break;
+	case BT_OGF_VS:
+		status = vs_cmd_put(cmd_in,
+				    raw_event_out,
+				    &return_param_length);
+		break;
+	default:
+		status = BT_HCI_ERR_UNKNOWN_CMD;
+		break;
+	}
+
+	if (!command_generates_command_complete_event(opcode) ||
+	    (status == BT_HCI_ERR_UNKNOWN_CMD))	{
+		encode_command_status(raw_event_out, opcode, status);
+	} else {
+		encode_command_complete_header(raw_event_out, opcode, return_param_length, status);
+	}
+}
+
+int hci_cmd_put(uint8_t *cmd_in)
+{
+	uint16_t opcode = *(uint16_t *)cmd_in;
+
+	if ((((struct bt_hci_cmd_hdr *)cmd_in)->param_len + BT_HCI_CMD_HDR_SIZE)
+		> HCI_CMD_PACKET_MAX_SIZE) {
+		return -EINVAL;
+	}
+
+	if (!IS_ENABLED(CONFIG_BT_CTLR_ADV_EXT)) {
+		cmd_put(cmd_in, &cmd_complete_or_status.raw_event[0]);
+	} else if (!is_host_using_legacy_and_extended_commands(opcode)) {
+		cmd_put(cmd_in, &cmd_complete_or_status.raw_event[0]);
+	} else {
+		/* The host is violating the specification
+		 * by mixing legacy and extended commands.
+		 */
+		if (command_generates_command_complete_event(opcode)) {
+			uint8_t len = CMD_COMPLETE_MIN_SIZE;
+			(void)encode_command_complete_header(cmd_complete_or_status.raw_event,
+							     opcode,
+							     len,
```

Comment:
Why not directly use CMD_COMPLETE_MIN_SIZE?

Reply:
- [ ] Use CMD_COMPLETE_MIN_SIZE directly